libs/providers/launchdarkly-client/src/lib/launchdarkly-client-provider.ts

```diff
@@ -34,73 +38,113 @@ export class LaunchDarklyClientProvider implements Provider {
     name: 'launchdarkly-client-provider',
   };
 
+  private readonly ldOptions: LDOptions | undefined;
   private readonly logger: Logger;
+  private _client?: LDClient;
+
+  public events = new OpenFeatureEventEmitter();
+
+  /*
+   * implement status field/accessor
+   * https://openfeature.dev/specification/sections/providers#requirement-242
+   * */
+  private _status: ProviderStatus = ProviderStatus.NOT_READY;
 
-  constructor(private client: LDClient, options: LaunchDarklyProviderOptions = {}) {
-    if (options.logger) {
-      this.logger = options.logger;
+  set status(status: ProviderStatus) {
+    this._status = status;
+  }
+
+  get status() {
+    return this._status;
+  }
+
+  constructor(
+    private readonly envKey: string,
+    private readonly initiaContext: LDContext,
+    { logger, ...ldOptions }: LaunchDarklyProviderOptions) {
+    if (logger) {
+      this.logger = logger;
     } else {
       this.logger = basicLogger({ level: 'info' });
     }
+    this.ldOptions = { ...ldOptions, logger: this.logger };
+
+  }
+
+  private get client(): LDClient {
+    if (!this._client) {
+      throw new GeneralError('Provider is not initialized');
+    }
+    return this._client;
   }
 
-  async initialize(context: EvaluationContext): Promise<void> {
-    const newContext = this.translateContext(context);
-    const oldContext = this.client.getContext();
-    const ignoreAnonymous = newContext.anonymous && oldContext.anonymous && !newContext.key
-    if(!ignoreAnonymous && !isEqual(newContext, oldContext)) {
-       await this.client.identify(newContext);
+  async initialize(): Promise<void> {
```

Comment:
I am potentially still missing something, but I think this takes a context, and my expectation is that context is used.


`  async initialize(context: EvaluationContext): Promise<void> {
	this._client = initialize(this.envKey, this.translateContext(evaluationContext), this.ldOptions);
`

Reply:
You are completely right I was overlooking an important sdk behaviour, I think it's alright now, could you check it out @kinyoklion